graphics/lvgl/Makefile

```
@@ -25,32 +25,67 @@ include $(APPDIR)/Make.defs
 LVGL_DIR = .
 LVGL_DIR_NAME = lvgl
 
-# Relax format check for LVGL to avoid errors on prinf() use
+# Relax LVGL's format checking and unused variable checking to avoid errors
 
-CFLAGS += -Wno-format
+CFLAGS += -Wno-format -Wno-unused-variable
 
-# LVGL Libraries
+-include ./lvgl/lvgl.mk
 
--include ./lvgl/src/lv_core/lv_core.mk
--include ./lvgl/src/lv_hal/lv_hal.mk
--include ./lvgl/src/lv_widgets/lv_widgets.mk
--include ./lvgl/src/lv_font/lv_font.mk
--include ./lvgl/src/lv_misc/lv_misc.mk
--include ./lvgl/src/lv_themes/lv_themes.mk
--include ./lvgl/src/lv_draw/lv_draw.mk
--include ./lvgl/src/lv_gpu/lv_gpu.mk
+CSRCS += port/lv_port.c
+CSRCS += port/lv_port_tick.c
 
-CSRCS += lv_tick_interface.c
 
-ifneq ($(CONFIG_USE_LV_FILESYSTEM),)
-CSRCS += lv_fs_interface.c
+ifneq ($(CONFIG_LV_PORT_USE_LCDDEV),)
+CSRCS += port/lv_port_lcddev.c
+endif
+
+ifneq ($(CONFIG_LV_PORT_USE_FBDEV),)
+CSRCS += port/lv_port_fbdev.c
+endif
+
+ifneq ($(CONFIG_LV_PORT_USE_TOUCHPAD),)
+CSRCS += port/lv_port_touchpad.c
+endif
+
+ifneq ($(CONFIG_LV_PORT_USE_BUTTON),)
+CSRCS += port/lv_port_button.c
+endif
+
+ifneq ($(CONFIG_LV_PORT_USE_KEYPAD),)
+CSRCS += port/lv_port_keypad.c
+endif
+
+ifneq ($(CONFIG_LV_PORT_USE_ENCODER),)
+CSRCS += port/lv_port_encoder.c
+endif
+
+ifneq ($(CONFIG_LV_PORT_USE_SYSLOG),)
+CSRCS += port/lv_port_syslog.c
+endif
+
+ifneq ($(CONFIG_LV_PORT_MEM_ATTRIBUTE_FAST_MEM_SECTION_NAME), "")
+CFLAGS   += "-DLV_ATTRIBUTE_FAST_MEM=__attribute__((section(CONFIG_LV_PORT_MEM_ATTRIBUTE_FAST_MEM_SECTION_NAME)))"
+CXXFLAGS += "-DLV_ATTRIBUTE_FAST_MEM=__attribute__((section(CONFIG_LV_PORT_MEM_ATTRIBUTE_FAST_MEM_SECTION_NAME)))"
```

```suggestion
CFLAGS   += "-DLV_ATTRIBUTE_FAST_MEM=locate_data(CONFIG_LV_PORT_MEM_ATTRIBUTE_FAST_MEM_SECTION_NAME)"
CXXFLAGS += "-DLV_ATTRIBUTE_FAST_MEM=locate_data(CONFIG_LV_PORT_MEM_ATTRIBUTE_FAST_MEM_SECTION_NAME)"
```
but most probably code will need to include `nuttx/compiler.h` so I'm not sure if this comment is applicable